### ISSUE TITLE
Confusion about ComplexNetwork

### ISSUE THREAD

**Issue by maergil** [reinforceio/tensorforce#471]
It seems that complex networks and layer networks do the same thing:

https://github.com/reinforceio/tensorforce/blob/master/tensorforce/core/networks/complex_network.py#L157 seems to say that

```python
ComplexNetwork([
  [a, b],
  [c],
]
```
is equivalent to

```python
ComplexNetwork([
  [a, b, c],
]
```
is equivalent to

```python
LayeredNetwork([a, b, c])
```
means

```python
output = c(b(a(input)))
```

Is this just a misunderstanding on my end?

**Comment by maergil** [comment]
https://github.com/reinforceio/tensorforce/blob/75a0629abfe9e7aa65e4ec3d6988b7953e5c3ad3/tensorforce/core/networks/complex_network.py#L157 that is

**Comment by AlexKuhnle** [comment]
Internally, layers are still maintained in a list, and indeed, without using the 'complex-network-specific' layer classes [Input](https://github.com/reinforceio/tensorforce/blob/75a0629abfe9e7aa65e4ec3d6988b7953e5c3ad3/tensorforce/core/networks/complex_network.py#L31) and [Output](https://github.com/reinforceio/tensorforce/blob/75a0629abfe9e7aa65e4ec3d6988b7953e5c3ad3/tensorforce/core/networks/complex_network.py#L111), it behaves the same as a normal layered network. These classes allow you to create additional / alternative links between layered subnetworks. Can provide an example later...

**Comment by Mazecreator** [comment]
@maergil  and @AlexKuhnle ,

Sorry, been away for a while with some personal issues, but just caught this issue.  I am not sure if the new structure of the TensorForce (recent major upgrades) affect the ComplexNetwork, but here are a few examples when the PR was pushed through:
https://github.com/reinforceio/tensorforce/pull/280

**Comment by maergil** [comment]
Thanks, that helps a lot! Maybe you could add these examples to the documentation? :)

**Comment by AlexKuhnle** [comment]
Hey @Mazecreator, I actually just looked a bit more into the complex network feature and realized that by now it's really just an extension of the normal `LayeredNetwork` class, so no need to treat it as a special case. Check out my latest commit, with minor breaking changes. Hope you don't mind the unification of these features, I think it works really well now.
@maergil, I changed our multi-state/action test to use the complex network feature now, [here](https://github.com/reinforceio/tensorforce/blob/master/tensorforce/tests/base_agent_test.py#L134), so this might help. If you have some lines of documentation with what you found helpful for explanation as PR, that would be great. ;-)
